core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/ScheduleGenerator.scala

```diff
@@ -75,10 +80,9 @@ abstract class ScheduleGenerator(
   def allocateResource(
       regionDAG: DirectedAcyclicGraph[Region, RegionLink]
   ): Unit = {
-    val dataTransferBatchSize = workflowContext.workflowSettings.dataTransferBatchSize
 
     val resourceAllocator =
-      new DefaultResourceAllocator(physicalPlan, executionClusterInfo, dataTransferBatchSize)
+      new DefaultResourceAllocator(physicalPlan, executionClusterInfo, workflowContext)
```

Comment:
I want to preserve `WorkflowContext` for other purpose. It is a mistake to have put `WorkflowSettings` inside `WorkflowContext`. Can we just pass `WorkflowSettings` into the allocator now? In the future we can refactor WorkflowSetting out of WorkflowContext.

Reply:
Yes. Fixed.